react/src/components/molecules/AccordionItem/index.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import SvgCircleChevron from '../../atoms/icons/SvgCircleChevron';
+import Collapse from '../../animations/Collapse';
+import './style.css';
+
+class AccordionItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({
+      open: !this.state.open
+    });
+  }
+
+  render() {
+    const buttonClasses = this.state.open ? 'ma__accordion-header__button is-open' : 'ma__accordion-header__button';
+    const accordionClasses = this.props.border ? 'ma__accordion-item' : 'ma__accordion-item__borderless';
+    const allowedChildren = ['Paragraph', 'OrderedList', 'UnorderedList', 'Heading', 'Table'];
+
+    return(
+      <div className={accordionClasses}>
+        <header className="ma__accordion-header">
+          <button
+            className={buttonClasses}
+            aria-label={this.props.info}
+            onClick={this.handleClick}
+            aria-expanded={this.state.open}
+          >
+            { this.props.icon && (
+              <div className="ma__accordion-header__icon">
+                {this.props.icon}
+              </div>
+            )}
+            <h2 className="ma__accordion-header__title">{this.props.title}</h2>
+          </button>
+        </header>
+        <Collapse in={this.state.open} dimension="height">
+          <div className="ma__accordion-content__body">
+            { React.Children.map(this.props.children, (child) => {
+              if (allowedChildren.includes(child.type.name)) {
```

Comment:
maybe should also check for  `React.isValidElement(child))` before `child.type.name`

Reply:
added that to the wrapper - we can remove the child check if that is prefered here